Source/Scene/Cesium3DTileset.js

```diff
@@ -964,52 +1064,32 @@ define([
                     }
                 }
                 Cesium3DTileOptimizations.checkChildrenWithinParent(tile3D, true);
-                if (tile3D.hasContent && hasEmptyChild && (tile3D.refine === Cesium3DTileRefine.REPLACE)) {
-                    // Tiles that use replacement refinement and have empty child tiles need to keep track of
-                    // descendants with content in order to refine correctly.
-                    refiningTiles.push(tile3D);
+
+                // Create a load heap, one for each unique server. We can only make limited requests to a given
+                // server so it is unecessary to keep a queue of all tiles needed to be loaded.
+                // Instead of creating a list of all tiles to load and then sorting it entirely to find the best ones,
+                // we keep just a heap so we have the best `maximumRequestsPerServer` to load. The order of these does
+                // not matter much as we will try to load them all.
+                // The heap approach is a O(n log k) to find the best tiles for loading.
+                var requestServer = tile3D.requestServer;
+                if (defined(requestServer)) {
+                    if (!defined(that._loadHeaps[requestServer])) {
+                        var heap = new Heap(sortForLoad);
+                        that._loadHeaps[requestServer] = heap;
+                        heap.maximumSize = RequestScheduler.maximumRequestsPerServer;
+                        heap.reserve(heap.maximumSize);
```

Comment:
Hm, not necessarily. Maximum size is only used to splice off extra elements should you exceed it, but it doesn't always have to be used. Reserve resizes the internal array. But you could increase the maximum size without ever reserving anything. In this case we know exactly how much space we're going to use, and it's not a lot anyway. There could be situations where there is a _potentially_ very, very large heap without knowing how much of it we'll need.